lib/skate/detours/trip_modification_publisher.ex

```diff
@@ -75,6 +75,29 @@ defmodule Skate.Detours.TripModificationPublisher do
     )
   end
 
+  @doc """
+  Publishes a `Realtime.Shape` to the configured `MQTT` server.
+
+  MQTT is optional for Skate, and callers should remember to handle both the
+  `:ok` and `:error` return values
+  """
+  def publish_shape(
+        %Realtime.Shape{} = shape,
+        opts \\ []
+      ) do
+    server = Keyword.get(opts, :server, @default_name)
+
+    GenServer.call(
+      server,
+      {
+        :new_shape,
+        %{
+          shape: shape
+        }
+      }
+    )
+  end
+
```

Comment:
issue: Given [`shape_id` is a required part of selected trips](https://gtfs.org/realtime/reference/#message-selectedtrips), I don't like publishing these separately and in two calls, it's fine if that's internal to this module, but IMO, I don't think callers should have the option. I get that this would potentially result in publishing the shape multiple times with the same data, but I don't think it matters, and I'd rather solve that at a later point when/if it becomes an issue.

I would rather see `%Realtime.Shape{}` become an argument to `publish_modification`.

Also, right now, you'll have a different ID for the Shape and for the Trip Modification, which also seems incorrect.

Reply:
> Also, right now, you'll have a different ID for the Shape and for the Trip Modification, which also seems incorrect.

Heh I think https://github.com/mbta/skate/pull/2689 resolves ☝️  (unless I'm misunderstanding your comment here)

---

> I don't like publishing these separately and in two calls

Yeah that's fair. I was of mixed minds about how this should be done, so I appreciate the nudge in that direction! Will update

Reply:
> > Also, right now, you'll have a different ID for the Shape and for the Trip Modification, which also seems incorrect.
> 
> Heh I think #2689 resolves ☝️ (unless I'm misunderstanding your comment here)

I'm thinking of the [generated ID in the `handle_call`](https://github.com/mbta/skate/pull/2688/files/055bf3ac369ecba1fe07fe83ac92bafd58701ac7#diff-9573b7338e9a4517a639fc5d4c6aab49d95b8334e56717f41aaddbbdd9729786R175)

Reply:
Oohhhh.

Question - in the medium-long-term, are those ID's supposed to correspond to, say, the trip modification ID, rather than being generated inside of the MQTT publisher?

Reply:
> are those ID's supposed to correspond to, say, the trip modification ID

"Trip Modification ID" is not a concept in the current Trip Modifications spec, so we're in charge of deciding what the ID of a trip modification is for internal communications and synchronization and it will not make it into GTFS-RT.

If eventually there is a spec which add's an ID, I could see the topic corresponding to the ID within the message.

As for whether the Trip Modification Publisher _should_ be responsible for generating ID's, I'm not sure. I like that it can generate it for us, but that poses the need for a new function[^1], that basically does the same thing as `publish_modification` but with an ID instead. 

[^1]: `publish_update`?

Reply:
I posed my question poorly, but that makes sense. Basically, we'll definitely need a version of this that takes an ID at some point, but we don't necessarily need that right now.